packages/prediction-app/src/features/model-templates-overview/components/ModelTemplateConfigForm/ModelTemplateConfigForm.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { useState, useMemo } from 'react'
+import {
+    Button,
+    ButtonStrip,
+    InputField,
+    Label,
+    SingleSelect,
+    SingleSelectOption,
+} from '@dhis2/ui'
+import i18n from '@dhis2/d2-i18n'
+import { useForm, Controller, useFieldArray } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+import { useModelTemplates } from '../../../../hooks/useModelTemplates'
+import { useRoute } from '../../../../hooks/useRoute'
+import { UserOptionsFields } from '../UserOptionsFields'
+import styles from './ModelTemplateConfigForm.module.css'
+
+const modelTemplateConfigSchema = z.object({
+    name: z.string().min(1, { message: i18n.t('Name is required') }),
+    templateId: z.number().min(1, { message: i18n.t('Template is required') }),
+    covariates: z.array(
+        z.object({
+            name: z.string()
+                .min(1, { message: i18n.t('Covariate name is required') })
+                .regex(/^\S*$/, { message: i18n.t('Covariate name cannot contain spaces') })
+        })
+    ),
+    userOptions: z.record(z.any())
+})
+
+export type ModelTemplateConfigFormValues = z.infer<typeof modelTemplateConfigSchema>
+
+interface ModelTemplateConfigFormProps {
+    onSubmit: (data: ModelTemplateConfigFormValues) => void
+    isSubmitting: boolean
+}
+
+export const ModelTemplateConfigForm = ({
+    onSubmit,
+    isSubmitting,
+}: ModelTemplateConfigFormProps) => {
+    const { route } = useRoute()
+    const { modelTemplates, isLoading: isLoadingTemplates } = useModelTemplates({ route })
+    const [newCovariate, setNewCovariate] = useState('')
+    const [covariateError, setCovariateError] = useState('')
+
+    const {
+        control,
+        handleSubmit,
+        formState: { errors },
+        watch,
+        setValue,
+    } = useForm<ModelTemplateConfigFormValues>({
+        resolver: zodResolver(modelTemplateConfigSchema),
+        defaultValues: {
+            name: '',
+            templateId: 0,
+            covariates: [],
+            userOptions: {}
+        }
+    })
```

Comment:
Perhaps you can extract this logic into a separate hook called `useFormController`? Place it inside ModelTemplateConfigForm/hooks. Also include the zod schema etc.